cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CreateNativeImageTask.java

```diff
@@ -0,0 +1,156 @@
+import java.util.Optional;
+
+import static io.ballerina.cli.launcher.LauncherUtils.createLauncherException;
+import static io.ballerina.cli.utils.FileUtils.getFileNameWithoutExtension;
+import static io.ballerina.projects.util.ProjectConstants.BLANG_COMPILED_JAR_EXT;
+import static io.ballerina.projects.util.ProjectConstants.USER_DIR;
+
+/**
+ * Task for creating the native image file.
+ *
+ * @since 2.0.0
+ */
+public class CreateNativeImageTask implements Task {
+    private final transient PrintStream out;
+    private Path output;
+    private Path currentDir;
+
+    public CreateNativeImageTask(PrintStream out, String output) {
+        this.out = out;
+        if (output != null) {
+            this.output = Paths.get(output);
+        }
+    }
+
+    @Override
+    public void execute(Project project) {
+        this.out.println();
+        this.out.println("Generating executable with native-image");
+        this.currentDir = Paths.get(System.getProperty(USER_DIR));
+
+        Path nativeDirectoryPath;
+        String nativeImageName;
+        Target target;
+
+        try {
+            if (project.kind().equals(ProjectKind.BUILD_PROJECT)) {
+                target = new Target(project.targetDir());
+            } else {
+                target = new Target(Files.createTempDirectory("ballerina-cache" + System.nanoTime()));
+                target.setOutputPath(getExecutablePath(project));
+            }
+        } catch (IOException e) {
+            throw createLauncherException("unable to resolve target path : " + e.getMessage());
+        } catch (ProjectException e) {
+            throw createLauncherException("unable to create executable : " + e.getMessage());
+        }
+
+        Path executablePath;
+        try {
+            executablePath = target.getExecutablePath(project.currentPackage()).toAbsolutePath().normalize();
+        } catch (IOException e) {
+            throw createLauncherException("unable to resolve executable path : " + e.getMessage());
+        }
+
+        if (project.kind().equals(ProjectKind.SINGLE_FILE_PROJECT)) {
+            Path sourceRootParent = Optional.of(project.sourceRoot().toAbsolutePath().getParent()).orElseThrow();
+            nativeDirectoryPath = sourceRootParent.resolve("native");
+            nativeImageName = Optional.of(sourceRootParent.getFileName()).orElseThrow().toString();
+        } else {
+            nativeDirectoryPath = target.path().toAbsolutePath().resolve("native");
+            nativeImageName = project.currentPackage().packageName().toString();
+        }
+
+        try {
+            // Create native directory
+            if (!Files.exists(nativeDirectoryPath)) {
+                Files.createDirectory(nativeDirectoryPath);
+            }
+
+            String[] command = {
+                    "native-image",
+                    "-jar",
+                    executablePath.toString(),
+                    nativeDirectoryPath + File.separator + nativeImageName,
```

Comment:
Without specifying the output path we can use the following two flags :
`-H:Name=<FileName>`: the name of the executable file that is generated(Default is the name of the jar file).
`-H:Path=<FileSystemPath>`: the directory where the generated executable is placed(If the directory does not exit then this will create the directory).

So we can only add `-H:Path=nativeDirectoryPath` and skip creating the directory. (Assuming the jar file name is same as the package name)

Reply:
Changed as suggested